pkg/providers/ovirt/templates/template-finder.go

```diff
@@ -0,0 +1,86 @@
+package templates
+
+import (
+	"fmt"
+	"strings"
+
+	templatev1 "github.com/openshift/api/template/v1"
+	tempclient "github.com/openshift/client-go/template/clientset/versioned/typed/template/v1"
+	ovirtsdk "github.com/ovirt/go-ovirt"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	templateNamespace = "openshift"
+	defaultLinux      = "rhel8"
+	defaultWindows    = "windows"
+)
+
+var osInfo = map[string]string{
+	"Red Hat Enterprise Linux Server": "rhel",
+	// TODO add more
+}
+
+// TemplateFinder attempts to find a template based on given parameters
+type TemplateFinder struct {
+	client *tempclient.TemplateV1Client
+}
+
+// NewTemplateFinder creates new TemplateFinder
+func NewTemplateFinder(tempClient *tempclient.TemplateV1Client) *TemplateFinder {
+	return &TemplateFinder{
+		client: tempClient,
+	}
+}
+
+// FindTemplate attempts to find best match for a template based on the source VM
+func (f *TemplateFinder) FindTemplate(vm *ovirtsdk.Vm) (*templatev1.Template, error) {
+	os := findOperatingSystem(vm)
+	workload := getWorkload(vm)
+	return f.getTemplate(os, workload)
+}
+
+func findOperatingSystem(vm *ovirtsdk.Vm) string {
+	if gos, found := vm.GuestOperatingSystem(); found {
+		distribution, _ := gos.Distribution()
+		version, _ := gos.Version()
+		fullVersion, _ := version.FullVersion()
+		return fmt.Sprintf("%s%s", osInfo[distribution], fullVersion)
```

Comment:
is the `fullVersion` always available? Can it be also a string incompatible with `osinfo-query`?

Reply:
My understanding is that it is always available. It is just a version and yes it could be incompatible with `os-info`. oVirt do not expose `os-info` so it is the best case we can have.